libraries/kotlin.test/common/src/main/kotlin/kotlin/test/Assertions.kt

```diff
@@ -72,6 +73,31 @@ fun <@OnlyInputTypes T> assertNotSame(illegal: T, actual: T, message: String? =
     asserter.assertNotSame(message, illegal, actual)
 }
 
+/**
+ * Asserts that [value] is of type [T], with an optional [message].
+ * 
+ * Note that due to type erasure the type check may be partial (e.g. assertIs<List<String>>(value)
+ * only checks for the class being [List] and not the type of its elements because it's erased).
+ */
+@SinceKotlin("1.5")
+@InlineOnly
+inline fun <reified T> assertIs(value: Any?, message: String? = null) {
+    contract { returns() implies (value is T) }
+    asserter.assertTrue({ messagePrefix(message) + "Expected value to be of type <${typeOf<T>()}>, actual <${value::class}>." }, value is T)
```

Comment:
Please ensure that the library compiles and its tests pass. `coreLibsTest` gradle task can be used for that.
In particular, `value::class` won't compile for nullable `value` because `::class` requires a non nullable LHS argument. We can use `value?.let { it::class }` instead.

Reply:
Yes of course. Sorry I wasn't done yet— just wanted to get the initial improvements in